addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/ForbiddenBypassScanRule.java

```diff
@@ -0,0 +1,179 @@
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+    }
+
+    @Override
+    public void scan() {
+        HttpMessage message = getNewMsg();
+        try {
+            sendAndReceive(message);
+            if (message.getResponseHeader().getStatusCode() != HttpStatusCode.FORBIDDEN) {
+                return;
+            }
+            URI uri = message.getRequestHeader().getURI();
+            String host = uri.getEscapedAuthority();
+            String path = uri.getEscapedPath();
+            String schema = uri.getScheme();
+            if (sendPathPayloads(path, host, schema, uri)) {
+                return;
+            }
+            if (sendHeaderPayloads(path, host, schema, uri)) {
+                return;
+            }
+
+        } catch (IOException e) {
+            LOG.warn(e.getMessage(), e);
+        }
+    }
+
+    private boolean sendPathPayloads(String path, String host, String schema, URI uri)
+            throws IOException {
+        String[] pathPayloads = {
+            "/%2e" + path,
+            path + "/./",
+            "/." + path + "/./",
+            path + "%20/",
+            "/%20" + path + "%20/",
+            path + "..;/",
+            path + "?",
+            path + "??",
+            "/" + path + "//",
+            path + "/",
+            path + "/.testus",
+            path + "../app.py"
+        };
+
+        for (String pathPayload : pathPayloads) {
+            HttpMessage reqWithPayload =
+                    new HttpMessage(new URI(schema + "://" + host + "" + pathPayload, true));
+            sendAndReceive(reqWithPayload);
+            if (reqWithPayload.getResponseHeader().getStatusCode() == HttpStatusCode.OK) {
+                createAlert(uri.toString(), reqWithPayload, pathPayload).raise();
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean sendHeaderPayloads(String path, String host, String schema, URI uri)
+            throws IOException {
+        String[] headerPayloads = {
+            "X-Rewrite-URL: " + path,
+            "X-Original-URL: " + path,
+            "Referer: " + path,
+            "X-Custom-IP-Authorization: 127.0.0.1",
+            "X-Originating-IP: 127.0.0.1",
+            "X-Forwarded-For: 127.0.0.1",
+            "X-Remote-IP: 127.0.0.1",
+            "X-Client-IP: 127.0.0.1",
+            "X-Host: 127.0.0.1",
+            "X-Forwared-Host: 127.0.0.1"
+        };
+
+        for (String header : headerPayloads) {
+            String tmpUri = schema + "://" + host;
+
+            if (header.contains("X-Rewrite-URL")) {
+                tmpUri = tmpUri + "/anything";
+            } else if (header.contains("X-Original-URL")) {
+                tmpUri = tmpUri + "/";
+            } else if (header.contains("Referer")) {
```

Comment:
This is now the same case as the first if statement, better include this condition there.